uitest/Dialogs/AddWidgetDialog.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) Microsoft Corporation and Contributors
+// Licensed under the MIT license.
+
+using DevHome.UITest.Extensions;
+using DevHome.UITest.Pages;
+using OpenQA.Selenium.Appium;
+using OpenQA.Selenium.Appium.Windows;
+
+namespace DevHome.UITest.Dialogs;
+
+/// <summary>
+/// Dialog model class for the add widget dialog
+/// </summary>
+public class AddWidgetDialog : PageDialog<DashboardPage>
+{
+    private WindowsElement SSHNavigationItem => Driver.FindElementByAccessibilityId($"{Configuration.Widget.IdPrefix}!App!!{Configuration.Widget.Provider}!!SSH_Wallet");
+
+    private WindowsElement MemoryNavigationItem => Driver.FindElementByAccessibilityId($"{Configuration.Widget.IdPrefix}!App!!{Configuration.Widget.Provider}!!System_Memory");
+
+    private WindowsElement NetworkUsageNavigationItem => Driver.FindElementByAccessibilityId($"{Configuration.Widget.IdPrefix}!App!!{Configuration.Widget.Provider}!!System_NetworkUsage");
+
+    private WindowsElement GPUUsageNavigationItem => Driver.FindElementByAccessibilityId($"{Configuration.Widget.IdPrefix}!App!!{Configuration.Widget.Provider}!!System_GPUUsage");
+
+    private WindowsElement CPUUsageNavigationItem => Driver.FindElementByAccessibilityId($"{Configuration.Widget.IdPrefix}!App!!{Configuration.Widget.Provider}!!System_CPUUsage");
+
+    private WindowsElement PinButton => Driver.FindElementByAccessibilityId("PinButton");
+
+    public AddWidgetDialog(WindowsDriver<WindowsElement> driver, DashboardPage dashboardPage)
+        : base(driver, dashboardPage)
+    {
+    }
+
+    public DashboardPage.WidgetControl AddMemoryWidget() => QuickAddWidget(MemoryNavigationItem);
+
+    public DashboardPage.WidgetControl AddNetworkUsageWidget() => QuickAddWidget(NetworkUsageNavigationItem);
+
+    public DashboardPage.WidgetControl AddGPUUsageWidget() => QuickAddWidget(GPUUsageNavigationItem);
+
+    public DashboardPage.WidgetControl AddCPUUsageWidget() => QuickAddWidget(CPUUsageNavigationItem);
+
+    public DashboardPage.WidgetControl AddSSHWidget(string configFilePath)
+    {
+        return WaitForWidgetToBeAdded(() =>
+        {
+            SSHNavigationItem.Click();
+
+            // Wait for the widget to be rendered before configuring and
+            // pinning it
+            // TODO: Can we use AccessibilityId for adaptive cards forms?
```

Comment:
Can we use AccessibilityId for adaptive cards forms?

CC: @krschau @ujjwalchadha

Reply:
What is the concern with using AccessibilityId? That it may change if Adaptive Cards gets updated? Or that we don't know the ID?
If it's about the later, I believe you can find it with Accessibility Insights or Inspect

Reply:
I've extracted the class name from the Accessibility Insights tool since I couldn't find the AutomationId. Both concerns are correct, if the JSON template changes in the future we need to reuse the Accessibility Insights tool to inspect the components that were rendered from the JSON template file and re-find the elements. Ideally, controlling the automation/accessibility ID is more forward compatible in case the template was modified in the future.